policyengine_core/simulations/simulation.py

```diff
@@ -704,10 +697,14 @@ def _run_formula(
                     )
             return values
 
-        if self.trace:
-            parameters_at = self.trace_parameters_at_instant
-        else:
-            parameters_at = self.tax_benefit_system.get_parameters_at_instant
+        if self.trace and not isinstance(
+            self.tax_benefit_system.parameters, TracingParameterNodeAtInstant
+        ):
+            # Soft-recast
+            self.tax_benefit_system.parameters.branch_name = self.branch_name
+            self.tax_benefit_system.parameters.trace = True
+            self.tax_benefit_system.parameters.tracer = self.tracer
+        parameters_at = self.tax_benefit_system.parameters
```

Comment:
By default, the `parameters` in `person`, `period`, `parameters` isn't actually the root of the parameter tree. Instead, it's a function that behaves like it (i.e. when you call it with a time period, it'll slice at that instant). This wastes computational time, because the slicing happens from the root all the way down - this was particularly wasteful in the SNAP EA calls: each month, recalculating every single parameter in the entire tree.